azuredevops/internal/service/build/resource_resource_authorization.go

```diff
@@ -33,17 +34,23 @@ func ResourceResourceAuthorization() *schema.Resource {
 			},
 			"resource_id": {
 				Type:         schema.TypeString,
-				Optional:     true,
+				Required:     true,
 				Description:  "id of the resource",
 				ValidateFunc: validation.NoZeroValues,
 			},
+			"definition_id": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Description:  "id of the build definition",
+				ValidateFunc: validation.NoZeroValues,
```

Comment:
`validate.NoEmptyStrings` 🙏 😁

Reply:
Just checked if the Terraform Plugin SDK provides something similar, so that we can remove code from our own code base. Not surprisingly there is `validation.StringIsNotWhiteSpace` 😁

Reply:
Oh my god! Gaaaaaaaaaaaaaaaah

Reply:
That's a type int, though soo, I think that's valid?

Reply:
I'll change the one above it, that's a string though

Reply:
Ha ... You paid attention! But with `NoZeroValue` you can set a negative value. Perhaps it's better to use `IntAtLeast(1)` or something.